packages/base-controller/src/ControllerMessenger.ts

```diff
@@ -1,30 +1,46 @@
-export type ActionHandler<Action, ActionType> = (
+export type ActionHandler<
+  Action extends ActionConstraint,
+  ActionType = Action['type'],
+> = (
   ...args: ExtractActionParameters<Action, ActionType>
 ) => ExtractActionResponse<Action, ActionType>;
-export type ExtractActionParameters<Action, T> = Action extends {
+
+export type ExtractActionParameters<
+  Action extends ActionConstraint,
+  T = Action['type'],
+> = Action extends {
   type: T;
-  handler: (...args: infer H) => any;
+  handler: (...args: infer H) => unknown;
 }
   ? H
   : never;
-export type ExtractActionResponse<Action, T> = Action extends {
+export type ExtractActionResponse<
+  Action extends ActionConstraint,
+  T = Action['type'],
+> = Action extends {
   type: T;
-  handler: (...args: any) => infer H;
+  handler: (...args: never[]) => infer H extends unknown;
```

Comment:
Highlighting to request closer review.

Reply:
Hmm. Using `never[]` doesn't seem right here; we want to allow arguments, so I wonder if we ought to use `unknown[]` instead. Also, can you explain why we would need `extends unknown` here? I feel like this is redundant.

Reply:
Removed the `unknown` thanks for pointing that out!

Reply:
Regarding `never[]`, see: https://github.com/MetaMask/core/pull/1890#issuecomment-1783050539